### PR TITLE
Added jid history, just like Sidekiq Enterprise

### PR DESCRIPTION
**Pull request by Akhilesh05**
I have added the enqueue time/jid history for the cron jobs. This feature is also present in _Sidekiq Ent_.
This will be quite a useful feature.
The default history size will be 10. User can also change it by adding `:cron_history_size` in their `sidekiq.yml` file.

**Have a look:**
image

### CODE REVIEW

File: lib/sidekiq/cron/job.rb [hunk 1]
```diff
@@ -58,19 +58,20 @@ def enque! time = Time.now.utc
...
           else
-            Sidekiq::Client.push(sidekiq_worker_message)
+            jid = Sidekiq::Client.push(sidekiq_worker_message)
           end
         end
 
         save_last_enqueue_time
+        add_jid_history jid
         logger.debug { "enqueued #{@name}: #{@message}" }
       end
```
Comment (jmettraux) [inline lib/sidekiq/cron/job.rb]:
```ruby

        jid =
          if klass_const
            if defined?(ActiveJob::Base) && klass_const < ActiveJob::Base
              enqueue_active_job(klass_const).try :provider_job_id
            else
              enqueue_sidekiq_worker(klass_const)
            end
          else
            if @active_job
              Sidekiq::Client.push(active_job_message)
            else
              Sidekiq::Client.push(sidekiq_worker_message)
            end
          end
```

File: lib/sidekiq/cron/job.rb [hunk 2]
```diff
@@ -370,6 +373,16 @@ def last_enqueue_time_from_redis
...
+      def jid_history_from_redis
+        out = nil
+        Sidekiq.redis do |conn|
+          out = conn.lrange(jid_history_key, 0, -1) rescue nil
+        end
+        out && out.map do |jid_history_raw|
+          Sidekiq.load_json jid_history_raw
+        end
+      end
+
```
Comment (jmettraux) [inline lib/sidekiq/cron/job.rb]:
```ruby
        out =
          Sidekiq.redis do |conn|
            conn.lrange(jid_history_key, 0, -1) rescue nil
          end
```

File: test/unit/web_extension_test.rb [hunk 3]
```diff
@@ -66,6 +66,25 @@ def app
...
 
+    it 'shows history of a cron job' do
+      @job.enque!
+      get "/cron/#{@name}"
+
+      jid = nil
+      Sidekiq.redis do |conn|
+        history = conn.lrange Sidekiq::Cron::Job.jid_history_key(@name), 0, -1
+        jid = Sidekiq.load_json(history.last)['jid']
+      end
```
Comment (jmettraux) [inline test/unit/web_extension_test.rb]:
```ruby
      jid =
        Sidekiq.redis do |conn|
          history = conn.lrange Sidekiq::Cron::Job.jid_history_key(@name), 0, -1
          Sidekiq.load_json(history.last)['jid']
        end
```

File: lib/sidekiq/cron/job.rb [hunk 4]
```diff
@@ -56,21 +56,23 @@ def enque! time = Time.now.utc
...
-        if klass_const
-          if defined?(ActiveJob::Base) && klass_const < ActiveJob::Base
-            enqueue_active_job(klass_const)
-          else
-            enqueue_sidekiq_worker(klass_const)
-          end
-        else
-          if @active_job
-            Sidekiq::Client.push(active_job_message)
+        jid =
```
Comment (ondrejbartas) [inline lib/sidekiq/cron/job.rb]: you are setting `jid = ` and then in every branch of condition you are doing the same.

if this it the case then why not to make method 

```ruby
def enqueue_job_based_on_type
         klass_const =
             begin
               Sidekiq::Cron::Support.constantize(@klass.to_s)
             rescue NameError
               nil
             end
  if klass_const
            if defined?(ActiveJob::Base) && klass_const < ActiveJob::Base
              enqueue_active_job(klass_const).try :provider_job_id
            else
              enqueue_sidekiq_worker(klass_const)
            end
          else
            if @active_job
              Sidekiq::Client.push(active_job_message)
            else
              Sidekiq::Client.push(sidekiq_worker_message)
            end
          end
end
```
and then just do `jid = enqueue_job_based_on_type`

**Comment by ondrejbartas** [comment]
@Akhilesh05 very nice work here, please if you would mind to do changes I am willing to merge it asap and release new version

**Comment by Akhilesh05** [comment]
@ondrejbartas You can merge it, It's a great feature to have the JID history
Thanks :smile:

**Comment by Akhilesh05** [comment]
Oh yes, Sure
I forgot the `jid =`
I'll make a commit with the corrections

**Comment by Akhilesh05** [comment]
@ondrejbartas Done :+1: 
You can now merge it
Thanks :smile:

**Comment by ondrejbartas** [comment]
@Akhilesh05 thanks for your PR. I will try to resolve another issues and then release new version. hopefully until end of week.

**Pull request merged**